cpp/daal/src/algorithms/pca/pca_dense_correlation_batch_impl.i

```diff
@@ -43,6 +43,27 @@ namespace internal
 using namespace daal::services::internal;
 using namespace daal::data_management;
 using namespace daal::internal;
+
+template <typename algorithmFPType, CpuType cpu>
+services::Status PCACorrelationKernel<batch, algorithmFPType, cpu>::computeSingularValues(const data_management::NumericTable & eigenvalues,
+                                                                                          data_management::NumericTable & singular_values,
+                                                                                          size_t nRows)
+{
+    const size_t nComponents = eigenvalues.getNumberOfColumns();
```

Comment:
Is it always equal to the number of columns?

Reply:
Yes, we do allocate eigenvalues array with (1 * component_count);